addon/adapters/interop-debug-adapter.js

```diff
@@ -2,18 +2,22 @@ import M3DebugAdapter from './m3-debug-adapter';
 import { get, defineProperty } from '@ember/object';
 import { inject } from '@ember/service';
 import { default as MegamorphicModel } from '../model';
-import require, { has } from 'require';
+import require from 'require';
 import { run } from '@ember/runloop';
+import DataAdapter from '@ember/debug/data-adapter';
+import { HAS_DEBUG_PACKAGE, HAS_EMBER_DATA_PACKAGE } from 'ember-m3/-infra/packages';
 
 let DebugAdapter;
-if (has('@ember-data/debug')) {
+if (HAS_DEBUG_PACKAGE) {
```

Comment:
It would be interesting to see if @eaf4's `@embroider/macro` library could satisfy these needs, if not we should reach out to him to ensure it does.

But I suspect it can today, via a combination of `moduleExists` and `importSync`. See: https://github.com/emberjs/rfcs/pull/507/files#diff-dee406d92e20c47ccea1291cc51ba8f8R550

Using the `@embroider/macros` is not a blocker, but knowing it can or cannot be used should be considered a blocker. If it cannot be used, let's articulate why, and work with @eaf4 to ensure in the future how ember-m3 / ember-data and embroider want to work, are compatible.

IMHO nothing here seems fundamentally incompatible, but I want to see if we can start either being compatible immediately, or informing parallel efforts to ensure we will soon be.

Reply:
For reference, the macro system can be used outside of embroider in classic apps as well via: https://www.npmjs.com/package/@embroider/macros

Reply:
it's not clear to me from reading those docs that `importSync` and some of the other macros are compatible outside of embroider given the extensive knowledge of the build they need. As the embroider ecosystem matures I'm happy to port to newer shared solutions, but given debug-macros is the standard in the community I'd prefer to stick to well trodden paths.

cc @rwjblue for thoughts.

Reply:
My recommendation is not to switch, my recommendation is to confirm this is going down that path that could be plausibly implemented with `@embroider/macros`. We want to avoid, going down a path that isn't aligned ecosystem wide.

My reading of the macros and this code suggests that the approaches are similar enough, that taking this approach today doesn't put us on a fundamentally incompatible path.

Reply:
But if this approach is being take, please be aware near term work will be requested to ensure this is compatible with embroider. So please prepare for that.